xlive/Blam/Engine/interface/screens/screen_xbox_live_task_progress_dialog.cpp

```diff
@@ -1,78 +1,56 @@
 #include "stdafx.h"
-
 #include "screen_xbox_live_task_progress_dialog.h"
-#include "H2MOD.h"
+
+#include "Blam/Engine/interface/user_interface_screen_widget_definition.h"
 #include "H2MOD/Tags/TagInterface.h"
 
-void c_xbox_live_task_progress_menu::Open(void* callback)
+void c_xbox_live_task_progress_menu::open(void* update_function)
 {
-	typedef void(__cdecl* c_screen_xbox_live_task_progress_dialog2)(int arg_0, signed int local_player_index, void* callback, int a4, int a5);
-	auto p_c_screen_xbox_live_task_progress_dialog2 = Memory::GetAddress<c_screen_xbox_live_task_progress_dialog2>(0x20C776);
+	typedef void(__cdecl* c_screen_xbox_live_task_progress_dialog_t)(int arg_0, signed int local_player_index, void* update_function, int a4, int a5);
+	auto p_c_screen_xbox_live_task_progress_dialog2 = Memory::GetAddress<c_screen_xbox_live_task_progress_dialog_t>(0x20C776);
 
-	p_c_screen_xbox_live_task_progress_dialog2(-1, 0, callback, 0, 0);
+	p_c_screen_xbox_live_task_progress_dialog2(-1, 0, update_function, 0, 0);
 }
 
-void c_xbox_live_task_progress_menu::ApplyPatches()
+void c_xbox_live_task_progress_menu::apply_patches()
 {
-	if (h2mod->GetEngineType() != _main_menu || Memory::IsDedicatedServer()) return;
```

Comment:
This line needs to be present there as you are not checking it inside `screens_apply_patches_on_map_load();` either

Reply:
The mainmenu check shouldn't matter since when it tries to find the tag datum on a map other than the mainmenu it'll return -1 for the task_progress_dialog_datum_index and return

Reply:
Yeah, but what is the point of going through all the tag paths just to find out the tag is not there? Literally a waste of CPU cycles and implicitly of electricity. Should add them back.

Reply:
moved the function so it doesn't execute unless its not a dedi and its the mainmenu engine